ioos_catalog/tasks/harvest.py

```diff
@@ -700,6 +716,69 @@ def erddap_geojson_url(self, coord_names):
         url_res.close()
         return gj
 
+
+    @classmethod
+    def get_time_from_dim(cls, time_var):
+        """Get min/max from a NetCDF time variable and convert to datetime"""
+        ndim = len(time_var.shape)
+        if ndim == 0:
+            ret_val = time_var.item()
+            res = ret_val, ret_val
+        elif ndim == 1:
+            # NetCDF Users' Guide states that when time is a coordinate variable,
+            # it should be monotonically increasing or decreasing with no
+            # repeated variables. Therefore, first and last elements for a
+            # vector should correspond to start and end time or end and start
+            # time respectively. See Section 2.3.1 of the NUG
+            res = time_var[0], time_var[-1]
+        else:
+            # TODO: use more efficient algorithm for finding max here
+            try:
+                if hasattr(time_var, 'calendar'):
+                    times = num2date(var[:], var.units, var.calendar)
+                else:
+                    times = num2date(var[:], var.units)
+                return np.min(times), np.max(times)
+            # need to add test
+            except ValueError:
+                return None, None
+
+        # if not > 1d, return the min and max elements found
+        min_elem, max_elem = np.min(res), np.max(res)
+        if hasattr(time_var, 'calendar'):
+            num2date([min_elem, max_elem], time_var.units,
+                      time_var.calendar)
+            return num2date([min_elem, max_elem], time_var.units,
+                            time_var.calendar)
+        else:
+            return num2date([min_elem, max_elem], time_var.units)
+
+
+
+    def get_min_max_time(self, cd):
+        """
+           Attempt to naively find a time variable in the dataset
+           and get the min/max
+        """
+        for v in cd._current_variables:
+            # we need a udunits time string in order for this to work
+            var = cd.nc.variables[v]
+            if hasattr(var, 'units'):
+                # assume this is time if 'since' is in the units string
+                # or this is the 'T' axis
```

Comment:
Just a heads up, this is not going to work in some cases. It's very possible that there are time variables that aren't dimensions with these units.

 ```
double time(time) ;
    time:standard_name = "time" ;
    time:units = "seconds since 1970-01-01" ;

double instrument_timestamp(time) ;
    time:standard_name = "time" ;
    time:units = "seconds since 1970-01-01" ;

double ingestion_timestamp(time) ;
    ingestion_timestamp:standard_name = "time" ;
    ingestion_timestamp:units = "seconds since 1970-01-01" ;
```

However, I think this will be good enough.